client/src/modules/accounts/accounts.service.js

```diff
@@ -52,8 +58,7 @@ function AccountService(Api, bhConstants) {
    *   an array of JSONs.
    */
   function read(id, options) {
-    return Api.read.call(this, id, options)
-      .then(handleAccounts);
+    return fetcher(id, options);
```

Comment:
It took me a minute to follow this logic because all of our other `read` methods follow the pattern: 
```
return fetchSomeData().then(processSomeData);
```

With this I had to follow the callback which then involved a good understanding of how `Function.prototype.call` worked. 

Could you move the `.then(handleAccounts)` out of the callback and into the `read` method? Then the callback and HttpCache are only responsible for loading accounts and the read method handles reading and pipes into processing. 

For example: 
```js
return fetcher(id, options)
  .then(handleAccounts);
```

Reply:
Yeah, good call.  I'll do that.